kube-runtime/src/watcher.rs

```diff
@@ -138,48 +150,89 @@ async fn step_trampolined<K: Resource + Clone + DeserializeOwned + Debug + Send
     state: State<K>,
 ) -> (Option<Result<Event<K>>>, State<K>) {
     match state {
-        State::Empty => match api.list(list_params).await {
-            Ok(list) => (Some(Ok(Event::Restarted(list.items))), State::InitListed {
-                resource_version: list.metadata.resource_version.unwrap(),
-            }),
-            Err(err) => (Some(Err(err).map_err(Error::InitialListFailed)), State::Empty),
-        },
-        State::InitListed { resource_version } => match api.watch(list_params, &resource_version).await {
-            Ok(stream) => (None, State::Watching {
-                resource_version,
-                stream: stream.boxed(),
-            }),
-            Err(err) => (
-                Some(Err(err).map_err(Error::WatchStartFailed)),
-                State::InitListed { resource_version },
-            ),
-        },
+        State::Empty {
+            continue_token,
+            mut objects,
+        } => {
+            let mut lp = list_params.clone();
+            lp.continue_token = continue_token;
+            lp.limit = Some(100);
```

Comment:
probably needs a parameter in `watcher::Config` to customise this

Reply:
67279a1a8d2083c93cecfbcf13d0e4a72cc21596

Reply:
looks great :+1: